src/Target/Php/Transpiler/BinaryOperation/BinaryOperationTranspiler.php

```diff
@@ -60,7 +63,16 @@ public function transpile(BinaryOperationNode $binaryOperationNode): string
         );
 
         $result = $expressionTranspiler->transpile($binaryOperationNode->operands->first);
-        $operator = sprintf(' %s ', $this->transpileBinaryOperator($binaryOperationNode->operator));
+
+        $binaryOperationTypeResolver = new BinaryOperationTypeResolver(scope: $this->scope);
+
+        if ($binaryOperationNode->operator === BinaryOperator::PLUS
+            && $binaryOperationTypeResolver->resolveTypeOf($binaryOperationNode)->is(StringType::get())
+        ) {
+            $operator = ' . ';
```

Comment:
Yes this is partially weird, i initially had the logic `checkIfAnyOfBinaryOperandNodesIsOfTypeString` in this class, but then rembered the `BinaryOperationTypeResolver` should implement it.
This way we make indirectly use of it